src/vcpkg/commands.applocal.cpp

```diff
@@ -0,0 +1,539 @@
+        {
+        }
+
+        void resolve(const Path& binary)
+        {
+            vcpkg::printf("vcpkg applocal processing: %s\n", binary);
+            auto dll_file = m_fs.open_for_read(binary, VCPKG_LINE_INFO);
+            const auto dll_metadata = vcpkg::try_read_dll_metadata(dll_file).value_or_exit(VCPKG_LINE_INFO);
+            const auto imported_names =
+                vcpkg::try_read_dll_imported_dll_names(dll_metadata, dll_file).value_or_exit(VCPKG_LINE_INFO);
+            Debug::print("Imported DLLs of ", binary, " were ", Strings::join("\n", imported_names), "\n");
+
+            for (auto&& imported_name : imported_names)
+            {
+                if (m_searched.find(imported_name) != m_searched.end())
+                {
+                    Debug::print("  %s: previously searched - Skip", imported_name);
+                    continue;
+                }
+                m_searched.insert(imported_name);
+
+                Path target_binary_dir = binary.parent_path();
+                Path installed_item_file_path = m_installed_bin_dir / imported_name;
+                Path target_item_file_path = Path(target_binary_dir) / imported_name;
+
+                if (m_fs.exists(installed_item_file_path, VCPKG_LINE_INFO))
+                {
+                    deploy_binary(m_deployment_dir, m_installed_bin_dir, imported_name);
+
+                    if (openni2_installed)
+                    {
+                        deployOpenNI2(target_binary_dir, m_installed_bin_parent, imported_name);
+                    }
+
+                    if (azurekinectsdk_installed)
+                    {
+                        deployAzureKinectSensorSDK(target_binary_dir, m_installed_bin_parent, imported_name);
+                    }
+
+                    if (magnum_installed)
+                    {
+                        bool g_is_debug = m_installed_bin_parent.stem() == "debug";
+                        if (g_is_debug)
+                        {
+                            deployMagnum(target_binary_dir, m_installed_bin_parent / "bin/magnum-d", imported_name);
+                        }
+                        else
+                        {
+                            deployMagnum(target_binary_dir, m_installed_bin_parent / "bin/magnum", imported_name);
+                        }
+                    }
+
+                    if (qt_installed)
+                    {
+                        deployQt(m_deployment_dir, m_installed_bin_parent / "plugins", imported_name);
+                    }
+
+                    resolve(m_deployment_dir / imported_name);
+                }
+                else if (m_fs.exists(target_item_file_path, VCPKG_LINE_INFO))
+                {
+                    Debug::print("  %s: %s not found in %s; locally deployed",
+                                 imported_name,
+                                 imported_name,
+                                 m_installed_bin_parent.c_str());
```

Comment:
`Debug::print()` concatenates -- it doesn't do `printf` formatting.
```suggestion
                    Debug::println("  ", imported_name, " not found in ", m_installed_bin_parent, "; locally deployed");
```

Reply:
Thanks, I have changed all occurrences